### PR TITLE
BI-5526: Add regexp_extract_all function

### PR DESCRIPTION
**Pull request by br0vk1n-ia**

### CODE REVIEW

**Comment by br0vk1n-ia** [comment]
Был вопросик по уровням вложенности, вывод у функции такой:
```
(SELECT array_agg(anon_2.strs) AS array_agg_1
  FROM (SELECT (REGEXP_MATCHES('100-200, 300-400', '(\d+)-(\d+)', 'g'))[1] AS strs) AS anon_2) AS anon_1
```

**Comment by br0vk1n-ia** [comment]
postgresql, as in other tests, returns None

File: lib/dl_connector_clickhouse/dl_connector_clickhouse/formula/definitions/functions_string.py [hunk 1]
```diff
@@ -130,6 +130,12 @@
             ),
         ]
     ),
+    # regexp_extract_all
+    base.FuncRegexpExtractAll(
+        variants=[
+            V(D.CLICKHOUSE, lambda text, pattern: sa.func.extractAll(sa.func.assumeNotNull(text), pattern)),
```
Comment (MCPN) [inline lib/dl_connector_clickhouse/dl_connector_clickhouse/formula/definitions/functions_string.py:136]: https://clickhouse.com/docs/en/sql-reference/functions/functions-for-nulls#assumenotnull
kinda scared of "Returned values: An arbitrary value, if the input value is NULL". can we wrap the case of `text` being `NULL`?
also, definitely need a test for that

**Review by MCPN — CHANGES_REQUESTED** [review]

File: lib/dl_connector_clickhouse/dl_connector_clickhouse/formula/definitions/functions_string.py (hunk 1 above)
Comment (br0vk1n-ia) [reply lib/dl_connector_clickhouse/dl_connector_clickhouse/formula/definitions/functions_string.py:136]: this functions wraps null text to ""

Comment (br0vk1n-ia) [reply lib/dl_connector_clickhouse/dl_connector_clickhouse/formula/definitions/functions_string.py:136]: added test

**Review by MCPN — APPROVED** [review]

**Pull request merged**